nose/importer.py

```diff
@@ -1,4 +1,8 @@
-"""Implements an importer that looks only in specific path (ignoring
+"""Adapted from the CPython 3.11 imp.py code.
+Copyright (c) 2001-2023 Python Software Foundation; All Rights Reserved.
+Originally licensed under the PSLv2 and incorporated under the LGPL 2.1.
```

Comment:
I’d suggest placing this comment above the code derived from CPython, and having a clear delimiter of where it ends, as it currently might seem to suggest that all of the file is based on CPython, rather than the compatibility portion being based on CPython and the rest being based on nose.

Reply:
Updated

Reply:
This looks good to me! I’m not a lawyer so I can’t really comment definitively on the fine details of contracting the licence notice years or potential implications of the termination clause, but I appreciate you addressing the compliance concerns here quickly.

Just a heads up, I plan to get in touch with the Python Software Foundation to see if I can get them to make some kind of public statement about the termination clause. That’s not me wanting to target you but rather because I think strict termination clauses are harmful for FOSS and that it would be a good thing for the Python ecosystem if the PSF were to make a public statement that anyone who becomes aware of a licence violation regarding PSF‐owned code and takes steps to address it in a reasonable amount of time should be considered to have been automatically re‐granted the licence. Licensing is hard and I don’t think people should be unduly punished for mistakes if they make a good faith effort to come back into compliance.